test/routes/files_test.js

```diff
@@ -0,0 +1,75 @@
+let get = require('../get');
+let fs = require('mz/fs');
+let path = require('path');
+let restler = require('restler');
+
+function upload(url, file) {
+  return new Promise((resolve, reject) => {
+    let req = restler.put(url, {
+      multipart: true,
+      data: {
+        upload: file
+      }
+    });
+
+    req.on('complete', (result, response) => {
+      result instanceof Error ?
+        reject(result) :
+        resolve(response);
+    });
+  });
+}
+
+async function file(location) {
+  let stats = await fs.stat(location);
+  return restler.file(location, null, stats.size, null, 'text/plain');
+}
+
+suite('GET /files', () => {
+  test('should succeed if file exists', async function() {
+    let res = await get(3000, '/files?filepath=/data/b2g/mozilla/profiles.ini');
+    res.statusCode.should.equal(200);
+    res.body.length.should.be.gt(0);
+  });
+
+  test('should fail if not found', async function() {
+    let res = await get(3000, '/files?filepath=/data/b2g/mozilla/wat');
+    res.statusCode.should.equal(404);
+  });
+
+  test('should fail if no file specified', async function() {
+    let res = await get(3000, '/files');
+    res.statusCode.should.equal(422);
+  });
+
+  test('should fail if file blank', async function() {
+    let res = await get(3000, '/files?filepath=');
+    res.statusCode.should.equal(422);
+  });
+});
+
+suite('PUT /files', () => {
+  let uploadFile;
+
+  setup(async function() {
+    uploadFile = await file(path.resolve(`${__dirname}/../fixtures/profiles.ini`));
+  });
+
+  test('should upload OK', async function() {
+    let url = 'http://localhost:3000/files?filepath=/data/b2g/mozilla/profiles.ini';
```

Comment:
__We__ probably don't need to in the test, but consumers should probably be URL encoding all their stuff. Right now I don't see anything that needs URL encoding. Thoughts?